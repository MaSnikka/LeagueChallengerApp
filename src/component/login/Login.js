import React, {Component} from 'react'
import { Redirect } from 'react-router-dom';
import {withRouter} from "react-router-dom";
import LoginForm from "./LoginForm";

class Login extends Component {

    constructor(props) {
        super(props);
    }

    handleRedirect = () => {
        this.props.history.push("/home");
    }

    render() {
        return (
            <div>
                <h1>Login</h1>
                <LoginForm loginRequested={this.handleRedirect}/>
            </div>
        )
    }
}

export default withRouter(Login);
