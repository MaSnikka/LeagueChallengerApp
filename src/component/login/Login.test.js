import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import Login from './Login';
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';

//Cleanup garbage from testing
afterEach(cleanup);

describe("<Login />", () => {
    //Makes snapshot of rendered HTML. If failed press U to update.
    it("Renders <Login /> component correctly", () => {
        const { asFragment } = render(<Router><Login /></Router>)
        expect(asFragment()).toMatchSnapshot();
    });

});