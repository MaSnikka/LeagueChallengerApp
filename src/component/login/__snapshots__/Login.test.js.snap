// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Renders <Login /> component correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Login
    </h1>
    <div>
      <div
        data-testid="login-form"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          placeholder="email"
          value=""
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password
        </label>
        <input
          placeholder="password"
          type="password"
          value=""
        />
      </div>
      <p />
      <div>
        <button
          data-testid="login-button"
          id="login-button"
        >
          Login
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
